src/capwap_http_api_handler.erl

```diff
@@ -20,408 +20,459 @@
-    end, {1, lists:duplicate(size(H), 0), []}, StatsIn),
-    Res1 = fmt_worker_stats(<<"total">>, Totals, Res),
-    {jsx:encode(Res1), Req, State};
+		  State = #s{verbose = Verbose}) ->
+	StatsIn = capwap_dp:get_stats(),
+	[H | _] = StatsIn,
+	{_, Totals, Res} = lists:foldl(fun(Stats, {Cnt, Sum, Acc}) ->
+						       S = tuple_to_list(Stats),
+						       NewAcc = case Verbose of
+									true ->
+										Label = bin_fmt("thread_~w", [Cnt]),
+										fmt_worker_stats(Label, S, Acc);
+									_  ->
+										Acc
+								end,
+						       {Cnt + 1, lists:zipwith(fun(X, Y) -> X + Y end, S, Sum), NewAcc}
+				       end, {1, lists:duplicate(size(H), 0), []}, StatsIn),
+	Res1 = fmt_worker_stats(<<"total">>, Totals, Res),
+	{jsx:encode(Res1), Req, State};
 handle_request_dp(_, _, Req, State) ->
-    {jsx:encode([{error, bad_command}]), Req, State}.
+	{jsx:encode([{error, bad_command}]), Req, State}.
 
 handle_request_metrics(text, Path, Req, State) ->
-    Metrics = lists:foldl(fun exo_entry_to_list/2, [],
-                          exometer:find_entries(Path)),
-    {prometheus_encode(Metrics), Req, State};
+	Metrics = lists:foldl(fun exo_entry_to_list/2, [],
+			      exometer:find_entries(Path)),
+	{prometheus_encode(Metrics), Req, State};
 handle_request_metrics(json, Path, Req, State) ->
-    Entries = lists:foldl(fun exo_entry_to_map/2, #{},
-                          exometer:find_entries(Path)),
-    Metrics = lists:foldl(fun(M, A) ->
-                                  maps:get(ioize(M), A) end, Entries,
-                          Path),
-    {jsx:encode(Metrics), Req, State}.
+	Entries = lists:foldl(fun exo_entry_to_map/2, #{},
+			      exometer:find_entries(Path)),
+	Metrics = lists:foldl(fun(M, A) ->
+					      maps:get(ioize(M), A) end, Entries,
+			      Path),
+	{jsx:encode(Metrics), Req, State}.
 
 %%%===================================================================
 %%% Internal functions
 %%%===================================================================
 get_release_vsn() ->
-    Releases = release_handler:which_releases(),
-    Vsn = case lists:keyfind("ergw-capwap-node", 1, Releases) of
-        false -> "none";
-        {_, Version, _, _} -> Version
-    end,
-    jsx:encode([{version, list_to_binary(Vsn)}]).
+	Releases = release_handler:which_releases(),
+	Vsn = case lists:keyfind("ergw-capwap-node", 1, Releases) of
+		      false -> "none";
+		      {_, Version, _, _} -> Version
+	      end,
+	jsx:encode([{version, list_to_binary(Vsn)}]).
 
 format_wtp({Id, Endpoint}) ->
-    [{id, Id}, {endpoint, fmt_endpoint(Endpoint)}].
+	[{id, Id}, {endpoint, fmt_endpoint(Endpoint)}].
 
 validate_hash(Hash) when length(Hash) == 64 ->
-    {ok, << <<(hex2dec(C)):4>> || C <- Hash >>}.
+	{ok, << <<(hex2dec(C)):4>> || C <- Hash >>}.
 
 hex2dec(C) when C >= $a andalso C =< $f -> C - $a + 10;
 hex2dec(C) when C >= $A andalso C =< $F -> C - $A + 10;
 hex2dec(C) when C >= $0 andalso C =< $9 -> C - $0.
 
-mac_to_bin(MAC) ->
```

Comment:
why did you remove this and replaced it with a much more complicated version (mac_to_hex)?